DICOM/src/Microsoft.Health.DeID.SharedLib/CryptoHashFunction.cs

```diff
@@ -8,67 +8,64 @@
 using System.Security.Cryptography;
 using System.Text;
 using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
 
 namespace Microsoft.Health.Dicom.DeID.SharedLib
 {
     public class CryptoHashFunction
     {
-        public static byte[] ComputeHmacSHA256Hash(byte[] input, byte[] hashKey = null)
+        private readonly HMAC _hmac;
+
+        public CryptoHashFunction(CryptoHashSetting cryptoHashSetting = null)
+        {
+            cryptoHashSetting ??= new CryptoHashSetting();
+            _hmac = cryptoHashSetting.CryptoHashKey == null ? new HMACSHA256() : new HMACSHA256(Encoding.UTF8.GetBytes(cryptoHashSetting.CryptoHashKey));
+        }
+
+        public byte[] ComputeHmacSHA256Hash(byte[] input)
         {
             if (input == null)
             {
                 return input;
             }
 
-            HMAC hmac = new HMACSHA256();
-            if (hashKey != null)
-            {
-                hmac = new HMACSHA256(hashKey);
-            }
-
-            return hmac.ComputeHash(input);
+            return _hmac.ComputeHash(input);
         }
 
-        public static byte[] ComputeHmacSHA256Hash(Stream input, byte[] hashKey = null)
+        public byte[] ComputeHmacSHA256Hash(Stream input)
```

Comment:
logic of ComputeHmacSHA256Hash for Stream and byte[] are identical. Could it be a generic function?
Not quite important since there're only two instances. But might be helpful if there're multiple instances.